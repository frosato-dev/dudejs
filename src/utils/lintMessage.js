import load from "@commitlint/load/lib/index";
import lint from "@commitlint/lint/lib/index";

export default async commitMessage => {
  const CONFIG = require("./../configs/commitlint");

  const options = await load(CONFIG);
  const report = await lint(commitMessage, options.rules);

  return report.errors;
};
