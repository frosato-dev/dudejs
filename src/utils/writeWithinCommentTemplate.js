export default commands => `
#### DudeJS - START ####
${commands.join("\n")}
#### DudeJS - END ####
`;
