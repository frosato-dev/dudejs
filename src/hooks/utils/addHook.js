import path from "path";
import fs from "fs";
import readPkgUp from "read-pkg-up";
import writeWithinCommentTemplate from "./../../utils/writeWithinCommentTemplate";

const modulePath = fs.realpathSync(process.cwd());
const clientProjectPath = path.join(modulePath, "../..");
const gitHookPath = path.join(clientProjectPath, ".git", "hooks");

export default (hookName, commands = []) => {
  const { pkg: packageJson } = readPkgUp.sync({ cwd: clientProjectPath });

  if (!packageJson) {
    //eslint-disable-next-line
    console.log(
      `‚ùå DudeJS was not able to add ${hookName} hook, come on dude did you forget to git init?`,
    );
    process.exit();
  }

  const filePath = path.join(gitHookPath, hookName);

  if (fs.existsSync(filePath)) {
    //eslint-disable-next-line
    console.log(
      `üòé Seems you already have a ${hookName} hook, DudeJS hasn't done anything and he likes it.`,
    );
    process.exit();
  }

  const fileContent = writeWithinCommentTemplate(commands);
  fs.writeFileSync(filePath, fileContent, { mode: 0o755 });

  //eslint-disable-next-line
  console.log(`‚úÖ DudeJS successfully installed ${hookName} hook for you.`);
};
