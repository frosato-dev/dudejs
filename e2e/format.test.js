import { exec, read, write, cleanSrcDirectory } from "./utils";
import { CLIENT_DIR } from "./constants";

const FILE_1_PATH = `${CLIENT_DIR}/src/toFormat-1.js`;
const FILE_2_PATH = `${CLIENT_DIR}/src/toFormat-2.js`;

beforeAll(() => {
  cleanSrcDirectory();
  write(FILE_1_PATH, `const hello = "Hello";`);
  write(FILE_2_PATH, `const addOne = (number) =>        number   + 1`);
});

it("should format all .js files under `src` directory", () => {
  exec("./node_modules/.bin/dudejs format");
  expect(read(FILE_1_PATH)).toMatchSnapshot("file 1 content");
  expect(read(FILE_2_PATH)).toMatchSnapshot("file 2 content");
});
