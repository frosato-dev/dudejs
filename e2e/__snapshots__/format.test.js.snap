// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should format all .js files under \`src\` directory: file 1 content 1`] = `
"const hello = 'Hello'
"
`;

exports[`should format all .js files under \`src\` directory: file 2 content 1`] = `
"const addOne = number => number + 1
"
`;
